tools/config.py

```diff
@@ -28,6 +65,69 @@ def read_file(path):
         error(f'Unable to read configuration file {path}!')
 
 
+def read_and_validate_config(path, required_config, log_file=None):
+    """Reads the configuration file and validates that all required sections
+       and options are defined.
+
+    Args:
+        path (string): path to the configuration file
+        required_config (dict): dictionary holding required {section, options}
+        log_file (fh): logging output to
+
+    Returns:
+        dict (str, dict): dictionary containing configuration settings
+    """
+    fn = sys._getframe().f_code.co_name
+
+    config = configparser.ConfigParser()
+    try:
+        config.read(path)
+    except Exception:
+        error(f"{fn}(): Unable to read configuration file {path}!")
+
+    # Check if all required configuration sections and options can be found.
+    # Also, process some values that are not just strings.
+    for section in required_config.keys():
+        if section not in config:
+            error(f"{fn}(): Missing section '{section}' in config file {path}.")
+        for item in required_config[section]:
+            if item not in config[section]:
+                error(f"{fn}(): Missing item '{item}' in section '[{section}]' "
```

Comment:
Rather than making this a hard error, you could also just pass through to a default value if the configuration setting is not specified?

```python
value = config[section].get(item, None)
if value is None:
    ...

Reply:
Possibly. Wonder if we can always specify a default value. But we can try.

If the configuration is only read at the beginning, crashing might be ok.